service/worker/scheduler/workflow.go

```diff
@@ -908,60 +887,57 @@ func (s *scheduler) startLongPollWatcher(ex *commonpb.WorkflowExecution) {
 	s.watchingWorkflowId = ex.WorkflowId
 }
 
-func (s *scheduler) cancelWorkflow(ex *commonpb.WorkflowExecution) error {
+func (s *scheduler) cancelWorkflow(ex *commonpb.WorkflowExecution) {
 	ctx := workflow.WithLocalActivityOptions(s.ctx, defaultLocalActivityOptions)
-	requestID, err := s.newUUIDString()
-	if err != nil {
-		return err
-	}
 	areq := &schedspb.CancelWorkflowRequest{
-		RequestId: requestID,
+		RequestId: s.newUUIDString(),
 		Identity:  s.identity(),
 		Execution: ex,
 		Reason:    "cancelled by schedule overlap policy",
 	}
-	err = workflow.ExecuteLocalActivity(ctx, s.a.CancelWorkflow, areq).Get(s.ctx, nil)
+	err := workflow.ExecuteLocalActivity(ctx, s.a.CancelWorkflow, areq).Get(s.ctx, nil)
 	if err != nil {
 		s.logger.Error("cancel workflow failed", "workflow", ex.WorkflowId, "error", err)
-		return err
 	}
 	// Note: the local activity has completed (or failed) here but the workflow might take time
 	// to close since a cancel is only a request.
-	return nil
+	// If this failed, that's okay, we'll try it again the next time we try to take an action.
 }
 
-func (s *scheduler) terminateWorkflow(ex *commonpb.WorkflowExecution) error {
+func (s *scheduler) terminateWorkflow(ex *commonpb.WorkflowExecution) {
 	ctx := workflow.WithLocalActivityOptions(s.ctx, defaultLocalActivityOptions)
-	requestID, err := s.newUUIDString()
-	if err != nil {
-		return err
-	}
 	areq := &schedspb.TerminateWorkflowRequest{
-		RequestId: requestID,
+		RequestId: s.newUUIDString(),
 		Identity:  s.identity(),
 		Execution: ex,
 		Reason:    "terminated by schedule overlap policy",
 	}
-	err = workflow.ExecuteLocalActivity(ctx, s.a.TerminateWorkflow, areq).Get(s.ctx, nil)
+	err := workflow.ExecuteLocalActivity(ctx, s.a.TerminateWorkflow, areq).Get(s.ctx, nil)
 	if err != nil {
 		s.logger.Error("terminate workflow failed", "workflow", ex.WorkflowId, "error", err)
 	}
-	return err
+	// Note: the local activity has completed (or failed) here but we'll still wait until we
+	// observe the workflow close (with a watcher) to start the next one.
+	// If this failed, that's okay, we'll try it again the next time we try to take an action.
 }
 
-func (s *scheduler) newUUIDString() (string, error) {
+func (s *scheduler) newUUIDString() string {
 	if len(s.uuidBatch) == 0 {
-		if err := workflow.SideEffect(s.ctx, func(ctx workflow.Context) interface{} {
+		panicIfErr(workflow.SideEffect(s.ctx, func(ctx workflow.Context) interface{} {
```

Comment:
To the worst, the worker will stuck in a crash loop?

Reply:
It's inside a workflow task.. the sdk will catch it and the workflow will be blocked